### DISCUSSION TITLE
Decryption by Venue Question

### DISCUSSION THREAD

**Discussion by unidesigner** [lucaapp/security-concept#39] [General]
Hi,

I may have over-looked this detail, but in the security concept it states that "luca then requests these venue owners to decrypt and upload the Check-In records of every user who visited their venue at the same time as the infected person."

Is it the case that the venue owners to _physically_ decrypt on-site on their own local server/app with their private keys, or is infrastructure from luca used? If infrastructure from luca is used this would mean that the private keys those private keys are accessible by luca at some point.

Thanks for the clarification.

**Comment by hrantzsch** [comment]
Hi,

correct, the venue owners perform the decryption locally on their own device. You can have a look into [this diagram](https://luca-app.de/securityconcept/processes/tracing_find_contacts.html#process). I think it provides a good overview of this process.

Kind regards

**Reply by unidesigner** [reply]
Thanks!
